packages/styled-components/src/constructors/constructWithOptions.ts

```diff
@@ -50,10 +50,7 @@ export interface Styled<
   <Props extends object = BaseObject, Statics extends object = BaseObject>(
     initialStyles: Styles<Substitute<OuterProps, NoInfer<Props>>>,
     ...interpolations: Interpolation<Substitute<OuterProps, NoInfer<Props>>>[]
-  ): IStyledComponent<R, Substitute<OuterProps, Props>> &
-    OuterStatics &
-    Statics &
-    (Target extends string ? {} : Target);
```

Comment:
we could change this to 
`  (R extends 'web' ? (Target extends string ? {} : Target) : {});`